backend/lib/use-cases/offices/office-assignees.test.ts

```diff
@@ -0,0 +1,138 @@
+import { ApplicationContext } from '../../adapters/types/basic';
+import * as factory from '../../factory';
+import OfficeAssigneesUseCase from './office-assignees';
+import { createMockApplicationContext } from '../../testing/testing-utilities';
+import { OfficeAssigneesRepository } from '../gateways.types';
+import { MockData } from '../../../../common/src/cams/test-utilities/mock-data';
+import { MOCKED_USTP_OFFICES_ARRAY } from '../../../../common/src/cams/offices';
+
+describe('OfficeAssigneesUseCase', () => {
+  let mockContext: ApplicationContext;
+  let mockRepo: OfficeAssigneesRepository;
+  let mockGateway: {
+    getOffices: jest.Mock;
+    getOfficeName: jest.Mock;
+  };
+  let getOfficeAssigneesRepositorySpy: jest.SpyInstance;
+  let getOfficesGatewaySpy: jest.SpyInstance;
+
+  beforeEach(async () => {
+    // Reset all mocks before each test
+    jest.clearAllMocks();
+
+    // Setup mock context
+    mockContext = await createMockApplicationContext();
+
+    // Setup mock repository
+    mockRepo = {
+      create: jest.fn(),
+      deleteMany: jest.fn(),
+      getDistinctAssigneesByOffice: jest.fn(),
+      search: jest.fn(),
+      release: jest.fn(),
+    };
+    getOfficeAssigneesRepositorySpy = jest
+      .spyOn(factory, 'getOfficeAssigneesRepository')
+      .mockReturnValue(mockRepo);
+
+    // Setup mock gateway
+    mockGateway = {
+      getOffices: jest.fn(),
+      getOfficeName: jest.fn(),
+    };
+    getOfficesGatewaySpy = jest.spyOn(factory, 'getOfficesGateway').mockReturnValue(mockGateway);
+  });
+
+  afterEach(() => {
+    // Restore all spies
+    getOfficeAssigneesRepositorySpy.mockRestore();
+    getOfficesGatewaySpy.mockRestore();
+  });
```

Comment:
It's a bit odd to clear mocks in the beforeEach and restore here. We should only need one (`restoreAllMocks()` is my preference).